cellengine/resources/attachment.py

```diff
@@ -1,58 +1,49 @@
 from __future__ import annotations
-from dataclasses import dataclass, field
-from dataclasses_json.cfg import config
+
+from attr import define, field
 
 import cellengine as ce
-from cellengine.utils.dataclass_mixin import DataClassMixin, ReadOnly
+from cellengine.utils import converter, readonly
 
 
-@dataclass
-class Attachment(DataClassMixin):
+@define
+class Attachment:
     """A class representing a CellEngine attachment.
     Attachments are non-data files that are stored in an experiment.
     """
 
+    _id: str = field(on_setattr=readonly)
     filename: str
-    _id: str = field(
-        metadata=config(field_name="_id"), default=ReadOnly()
-    )  # type: ignore
-    crc32c: str = field(repr=False, default=ReadOnly())  # type: ignore
-    experiment_id: str = field(repr=False, default=ReadOnly())  # type: ignore
-    md5: str = field(repr=False, default=ReadOnly())  # type: ignore
-    size: int = field(repr=False, default=ReadOnly())  # type: ignore
+    crc32c: str = field(on_setattr=readonly, repr=False)
+    experiment_id: str = field(on_setattr=readonly, repr=False)
+    md5: str = field(on_setattr=readonly, repr=False)
+    size: int = field(on_setattr=readonly, repr=False)
 
-    @classmethod
-    def get(cls, experiment_id: str, _id: str = None, name: str = None) -> Attachment:
-        """Get an Attachment by name or ID for a specific experiment. Either
-        `name` or `_id` must be specified.
+    @property
+    def client(self):
+        return ce.APIClient()
```

Comment:
Why add these `client` properties instead of just using `ce.APIClient()` in the methods that need it?

Reply:
I have a vague sense that maybe passing the client object around is better than using a `Singleton`, but maybe that's completely not worth worrying about

Reply:
The props increase the effective API surface, so I would get rid of them.